### PR TITLE
Auto-generate Flow definitions from TS definitions using flowgen

### PR DESCRIPTION
**Pull request by kitten**
I've been meaning to add this for a while; Using [flowgen](https://github.com/joarwilk/flowgen) we can generate some nice Flow definitions from our rather simple TS definition files.

This will still need to be tested but I'm already opening a PR so that it's visible.
The script that's being added generates `.js.flow` files for each `d.ts` file.

This also upgrades `bs-platform` which contains a rather important fix for arity-0 functions being passed when a user is meaning to ignore the only argument, which lead to an infinite loop before.

### CODE REVIEW

File: scripts/generate-flow-files.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  if (input.length === 0) {
+    throw new Error('No input files passed as arguments.');
+  }
+
+  console.log(`Compiling ${input.length} TS definitions to Flow...`);
+
+  const defs = input.map(filename => {
+    const fullpath = path.resolve(cwd, filename);
+
+    let flowdef = compiler.compileDefinitionFile(fullpath);
```
Comment (parkerziegler) [inline scripts/generate-flow-files.js]: Any reason not to just wrap this in `beautify` directly? `compileDefinitionFile` appears sync, as does `beautify`. The pipe operator would be nice here 😂 

```suggestion
    const flowdef = beautify(compiler.compileDefinitionFile(fullpath));
```

Comment (kitten) [reply scripts/generate-flow-files.js]: lol that was because I wasn't sure whether it was working, so I must've forgotten to clean it up after

**Comment by parkerziegler** [comment]
Really sweet! One question I have – is there a particular reason we don't want these checked in to version control? I've never used Flow 😱 and don't know much about what should / shouldn't be checked in to vc.

File: scripts/generate-flow-files.js (hunk 1 above)
Comment (parkerziegler) [reply scripts/generate-flow-files.js]: ^ Literally me on every PR (I do a lot of force pushing later on).

**Comment by kitten** [comment]
@parkerziegler There's really no particular reason _not_ to check them in, but I prefer not to check in build artifacts. As long as we assume there don't have to be any manual changes to the autogenerated `.js.flow` files, it's less error-prone to not check them in. Otherwise it's easy to forget to regenerate them, to make manual changes that'll then be overwritten, etc.

**Comment by cem2ran** [comment]
@kitten I'm curious if you tried out [genType](https://github.com/cristianoc/genType) and if so if you ran into any issues generating types for TypeScript and Flow through it?

**Comment by kitten** [comment]
@cem2ran yes, I've tried it before but ran into issues where the generated types weren't quite working as they're very much geared towards types for apps. I think I also encountered an issue when trying to then use it in another library or app that wasn't using gentype.

The issue here is still open so I'd say it's just not there yet: https://github.com/cristianoc/genType/issues/95

From a maintenance standpoint I believe though that it's completely acceptable to author Typescript definitions separately

**Pull request merged**